tests/ipsecondary_test.py

```diff
@@ -145,6 +145,40 @@ def test_secondary_ostree_update(uptane_repo, secondary, aktualizr, treehub, sys
     return True
 
 
+@with_treehub()
+@with_uptane_backend()
+@with_director()
+@with_sysroot()
+@with_secondary(start=False, output_logs=False, force_reboot=True)
+@with_aktualizr(start=False, run_mode='once', output_logs=True)
+def test_secondary_ostree_reboot(uptane_repo, secondary, aktualizr, treehub, sysroot, director, **kwargs):
+    target_rev = treehub.revision
+    expected_targetname = uptane_repo.add_ostree_target(secondary.id, target_rev, "GARAGE_TARGET_NAME")
+
+    with secondary:
+        with aktualizr:
+            aktualizr.wait_for_completion()
+        secondary.wait_for_completion()
+
+    pending_rev = aktualizr.get_current_pending_image_info(secondary.id)
+
+    if pending_rev != target_rev:
+        logger.error("Pending version {} != the target one {}".format(pending_rev, target_rev))
+        return False
+
+    sysroot.update_revision(pending_rev)
+
+    with secondary:
+        with aktualizr:
+            aktualizr.wait_for_completion()
+
+    if not director.get_install_result():
```

Comment:
I think it makes sense to add checking if what is currently installed on Secondary is what we expect
something like this `if target_rev != aktualizr.get_current_image_info(secondary.id):